estate/models/estate_property.py

```diff
@@ -0,0 +1,113 @@
+from dateutil.relativedelta import relativedelta
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.float_utils import float_compare, float_is_zero
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Estate property"
+    _order = "id desc"
+
+    name = fields.Char(string="Title", required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(string="Available From", default=fields.Date.today() + relativedelta(months=3), copy=False)
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer(string="Living Area (sqm)")
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer(string="Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        selection=[("north", "North"), ("south", "South"), ("east", "East"), ("west", "West")],
+    )
+
+    # Reserved fields
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        required=True,
+        selection=[("new", "New"), ("offer_received", "Offer Received"), ("offer_accepted", "Offer Accepted"), ("sold", "Sold"), ("cancelled", "Cancelled")],
+        default="new",
+        copy=False,
+        string="Status",
+    )
+
+    # Relations
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    salesman_id = fields.Many2one("res.users", default=lambda self: self.env.user)
+    buyer_id = fields.Many2one("res.partner", copy=False)
+    tag_ids = fields.Many2many("estate.property.tag")
+    offer_ids = fields.One2many("estate.property.offer", "property_id")
+
+    # Computed
+    total_area = fields.Integer(compute="_compute_total_area", string="Total Area (sqm)")
+    best_price = fields.Float(compute="_compute_best_price", string="Best Offer")
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self) -> None:
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids.price")
+    def _compute_best_price(self) -> None:
+        for record in self:
+            record.best_price = max(record.offer_ids.mapped("price"), default=0)
+
+    # Methods that trigger on changes
+    @api.onchange("garden")
+    def _compute_garden_defaults(self) -> None:
```

Comment:
```suggestion
    def _onchange_garden_defaults(self) -> None:
```
This is an `onchange`, not a `compute`. Should be named `_onchange_garden_defaults` or `_onchange_garden`.